sdk/eventhub/Azure.Messaging.EventHubs/MIGRATIONGUIDE.md

```diff
@@ -0,0 +1,393 @@
+
+In v5, events can be streamed as they come in without having to use a batched read approach.
+
+This code which reads from a partition in v4:
+
+```csharp
+var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
+var eventHubName = "<< NAME OF THE EVENT HUB >>";
+
+var connectionStringBuilder = new EventHubsConnectionStringBuilder(connectionString){ EntityPath = eventHubName }; 
+var eventHubClient = EventHubClient.CreateFromConnectionString(connectionStringBuilder.ToString());
+PartitionReceiver partitionReceiver = client.CreateReceiver("my-consumer-group", "my-partition-id", EventPosition.FromStart());
+try
+{
+    // Gets up to 100 events or until the read timeout elapses.
+    IEnumerable<EventData> eventDatas = await partitionReceiver.ReceiveAsync(100);
+    // Gets the next 50 events or until the read timeout elapses.
+    IEnumerable<EventData> eventDatas = await partitionReceiver.ReceiveAsync(50);
+}
+catch()
+{
+    // Handle the exception here
+}
+finally
+{
+    await partitionReceiver.CloseAsync();
+    await eventHubClient.CloseAsync();
+}
+```
+
+Becomes this in v5:
+```csharp
+var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
+var eventHubName = "<< NAME OF THE EVENT HUB >>";
+
+string consumerGroup = EventHubConsumerClient.DefaultConsumerGroupName;
+
+await using (var consumer = new EventHubConsumerClient(consumerGroup, connectionString, eventHubName))
+{
+    EventPosition startingPosition = EventPosition.Earliest;
+    string partitionId = (await consumer.GetPartitionIdsAsync()).First();
+
+    using var cancellationSource = new CancellationTokenSource();
+    cancellationSource.CancelAfter(TimeSpan.FromSeconds(45));
+
+    await foreach (PartitionEvent receivedEvent in consumer.ReadEventsFromPartitionAsync(partitionId, startingPosition, cancellationSource.Token))
+    {
+        // At this point, the loop will wait for events to be available in the partition.  When an event
+        // is available, the loop will iterate with the event that was read.  Because we did not
+        // specify a maximum wait time, the loop will wait forever unless cancellation is requested using
+        // the cancellation token.
+    }
+}
+```
+
+See [`ReadEvents.cs`](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample09_ReadEventsFromAKnownPosition.cs) for a sample program demonstrating this.
+
+### Migrating code from `EventProcessorHost` to `EventProcessorClient` for reading events
+
+In v4, `EventProcessorHost` allowed you to balance the load between multiple instances of your program and checkpoint
+events when receiving. Developers would have to create and register a concrete implementation of `IEventProcessor` to
+begin consuming events.
+
+In v5, `EventProcessorClient` allows you to do the same, the development model is made simpler by registering functions that would be invoked for each event. To use
```

Comment:
We should probably make an explicit statement about "... made simpler by using events.  Rather than implementing an interface, you subscribe to the events that you are interested in by registering an event handler delegate, following the standard .NET pattern."